src/main/java/com/dev/movietheatre/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.dev.movietheatre.service.impl;
+
+import com.dev.movietheatre.dao.ShoppingCartDao;
+import com.dev.movietheatre.dao.TicketDao;
+import com.dev.movietheatre.lib.Inject;
+import com.dev.movietheatre.lib.Service;
+import com.dev.movietheatre.model.MovieSession;
+import com.dev.movietheatre.model.ShoppingCart;
+import com.dev.movietheatre.model.Ticket;
+import com.dev.movietheatre.model.User;
+import com.dev.movietheatre.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
```

Comment:
if there is no shopping cart for the user - we still save ticket to the db, is it ok?

Reply:
I don't think so. When we register user, we register new shopping cart for him (Hibernate-06 Checklist. 5.Use method registerNewShoppingCart() in AuthenticationServiceImpl class in register() method.). So, we save ticket to existed shopping card.